Library/Formula/logtalk.rb

```diff
@@ -1,33 +1,54 @@
-require "formula"
-
 class Logtalk < Formula
   homepage "http://logtalk.org"
-  url "http://logtalk.org/files/lgt2441.tar.bz2"
-  sha256 "ca49d8e26a08dcf2a39358efa827789ecc5c47791035e5bb5830a6aec59b94e9"
-  version "2.44.1"
-
-  devel do
-    url "http://logtalk.org/files/logtalk-3.00.0-rc7.tar.bz2"
-    sha256 "50664e28eaa75eb0b293345919a9178bc9b2e2135a43f066265cfa99cb2c042a"
-    version "3.00.0-rc7"
-  end
+  url "http://logtalk.org/files/logtalk-3.00.0.tar.bz2"
+  sha256 "7462eaf8fd7fa865f8cccbb449a97981fbcb109a21536e2420f205b3b4b814b2"
 
   head "https://github.com/LogtalkDotOrg/logtalk3.git"
 
-  option "swi-prolog", "Build using SWI Prolog as backend"
-  option "gnu-prolog", "Build using GNU Prolog as backend (Default)"
+  option "with-swi-prolog", "Build using SWI Prolog as backend"
+  option "with-gnu-prolog", "Build using GNU Prolog as backend"
+  option "with-yap-prolog", "Build using YAP Prolog as backend (Default)"
+
+  deprecated_option "swi-prolog" => "with-swi-prolog"
+  deprecated_option "gnu-prolog" => "with-gnu-prolog"
 
-  if build.include?("swi-prolog")
+  if build.with?("swi-prolog")
     depends_on "swi-prolog"
-  else
+  elsif build.with?("gnu-prolog")
     depends_on "gnu-prolog"
+  else
+    depends_on "yap"
   end
 
   def install
-    if build.stable?
-      cd("scripts") { system "./install.sh", prefix }
-    else
-      cd("scripts") { system "./install.sh", "-p", prefix }
-    end
+    cd("scripts") { system "./install.sh", "-p", prefix }
+  end
+
+  def caveats; <<-EOS.undent
+    You must set the LOGTALKHOME and LOGTALKUSER environment variables before
+    using logtalk:
+
+      export LOGTALKHOME=#{share}/logtalk-#{version}-stable
+      export LOGTALKUSER=~/logtalk
+  EOS
+  end
+
+  test do
+    user_setup = testpath/"test-logtalk"
+    home = "#{share}/logtalk-#{version}-stable"
+    ENV["LOGTALKHOME"] = home
+    ENV["LOGTALKUSER"] = user_setup
+    system "#{bin}/logtalk_user_setup"
+    assert Dir.exist? user_setup
+
+    lgt = if build.with?("swi-prolog")
+            "swilgt"
+          elsif build.with?("gnu-prolog")
+            "gplgt"
+          else
+            "yaplgt"
+          end
+
+    pipe_output("#{lgt} #{home}/examples/ack/ack.lgt 2>/dev/null", "")
```

Comment:
`pipe_output` [makes no assertions](https://github.com/Homebrew/homebrew/blob/700635915dcd0cbc412cd7aab8aac177f3d8acbc/Library/Homebrew/formula_assertions.rb#L35-L41), it only returns the output of running the command.

For example, `pipe_output("exit 1", "")` succeeds.